lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,57 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after [timesToPump]  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
```

Comment:
We don't export `AsyncMatcher` because I don't want to commit to the API, especially since I'm planning a substantial refactor to the upstream `matcher` API. I don't want to declare the return type as something we don't export.

The documentation comments for async matchers should probably be updated to avoid mentioning `AsyncMatcher` directly, though.